src/zenml/orchestrators/utils.py

```diff
@@ -50,6 +51,35 @@ def create_tfx_pipeline(
     )
 
 
+def is_cached_step(execution_info: data_types.ExecutionInfo) -> bool:
+    """Returns the caching status of a tfx step.
+
+    Args:
+        execution_info: The execution info of a tfx step.
+
+    Returns:
+        The caching status of a tfx step as a boolean value.
+    """
+    repository = Repository()
+    metadata_store = repository.get_stack(
+        repository.active_stack_name
+    ).metadata_store
+
+    step_name = str(
+        execution_info.exec_properties["zenml-pipeline_parameter_name"]
+    )[1:-1]
```

Comment:
Why do we need to do `1:-1` here? If we want to strip it we can use `.strip()`? Also we should add exception handling here if this exec_property does not exist (Maybe raise a KeyError)

Reply:
You can't actually use `.strip()` on it. Prior to doing the slicing, we have `'"step_one"'`. For some reason `tfx` packages it that way. I'd be happy if there was a Python method that removed the extra quotation marks, but I don't know it. Any suggestions @htahir1?

Reply:
I think its a json string so you can do `import json; json.loads(execution_info.exec_properties[step_property_name])`

Reply:
You can optionally pass in characters to strip, so `.strip(")` would probably work in this case.
I do however agree that we should use `json.loads` instead, as we (and not actually tfx) put them in the metadata store using `json.dumps` in the first place.

Reply:
Done